src/android/CodePush.java

```diff
@@ -109,17 +113,17 @@ private boolean execUpdateSuccess(CallbackContext callbackContext) {
             /* save reporting status for first install */
             try {
                 String appVersion = Utilities.getAppVersionName(cordova.getActivity());
-                codePushReportingManager.reportStatus(CodePushReportingManager.Status.STORE_VERSION, null, appVersion, mainWebView.getPreferences().getString(DEPLOYMENT_KEY_PREFERENCE, null), this.mainWebView);
+                codePushReportingManager.reportStatus(new StatusReport(ReportingStatus.STORE_VERSION, null, appVersion, mainWebView.getPreferences().getString(DEPLOYMENT_KEY_PREFERENCE, null)), this.mainWebView);
             } catch (PackageManager.NameNotFoundException e) {
                 // Should not happen unless the appVersion is not specified, in which case we can't report anything anyway.
                 e.printStackTrace();
             }
-        }
-
-        if (this.codePushPackageManager.installNeedsConfirmation()) {
+        } else if (this.codePushPackageManager.installNeedsConfirmation()) {
```

Comment:
this appears to be a big logic change, from if to else if, what is the context behind this that it doesn't break the previous behavior?